packages/react-core/src/components/Modal/ModalContent.tsx

```diff
@@ -90,13 +93,16 @@ export const ModalContent: React.FunctionComponent<ModalContentProps> = ({
       title={title}
       id={modalBoxAriaDescribedById || id}
     >
-      {showClose && <ModalBoxCloseButton onClose={onClose} />}
-      {modalBoxHeader}
-      {description && <ModalBoxDescription id={id}>{description}</ModalBoxDescription>}
-      <ModalBoxBody {...props} {...(!description && { id })}>
-        {children}
-      </ModalBoxBody>
-      {modalBoxFooter}
+      {!isBasic && showClose && <ModalBoxCloseButton onClose={onClose} />}
```

Comment:
Can we remove this `isBasic` prop and unify all these elements into one component?
```JSX
const ModalBody = () => {
  onst modalBoxHeader = header ? (
    <div className={css(titleStyles.title)}>{header}</div>
  ) : (
    <ModalBoxHeader hideTitle={hideTitle}> {title} </ModalBoxHeader>
  );

  const modalBoxFooter = footer ? (
    <ModalBoxFooter>{footer}</ModalBoxFooter>
  ) : (
    actions.length > 0 && <ModalBoxFooter>{actions}</ModalBoxFooter>
  );
  return <Fragment>
    {showClose && <ModalBoxCloseButton onClose={onClose} />}
    {modalBoxHeader}
    {description && <ModalBoxDescription id={id}>{description}</ModalBoxDescription>}
    <ModalBoxBody {...props} {...(!description && { id })}>
          {children}
        </ModalBoxBody>
    {modalBoxFooter}
</Fragment>
}
```

This way we won't have a ton of ifs in this component and it can just have children as with any other component.

If we don't want to break backwards compatibility that much we could add prop called bodyWrapper to Modal where you'd by default use `ModalBody` component, if you'd don't want to use you could pass in `null` or w/e. The bodyWrapper prop could be either null, string, number, function or node.